src/libsystemd/sd-bus/bus-slot.c

```diff
@@ -180,6 +190,9 @@ void bus_slot_disconnect(sd_bus_slot *slot) {
                 assert_not_reached("Wut? Unknown slot type?");
         }
 
+        if (slot->call_cleanup && callback)
+                callback(NULL, slot->userdata, NULL);
```

Comment:
should be enclosed in (void) i guess...

More importantly though: this would not be called when the slot is freed but possibly much earlier, in case of "floating" slots, as in that case the slots will be disconnected from the bus when the bus is freed, and would then just exist "disconnected" until the last ref is dropped. This callback would be called when the disconnection happens, not when the slot is freed.